aws-android-sdk-mobile-client/src/androidTest/java/com/amazonaws/mobile/client/AWSMobileClientTest.java

```diff
@@ -686,6 +686,27 @@ public void testSignedOutWithoutRevokeToken() throws Exception {
     public void testSignInWrongPassword() throws Exception {
         AWSMobileClient.getInstance().signIn(getPackageConfigure().getString("username"), "wrong", null);
     }
+    
+    @Test
+    public void testDeleteUser() throws Exception {
+        auth.signIn(username, PASSWORD, null);
+        assertTrue("isSignedIn is true", auth.isSignedIn());
+        
+        auth.deleteUser();
+        assertTrue("isSignedIn is false", auth.isSignedIn());
+        
+        try {
+            auth.signIn(username, PASSWORD, null);
+            fail("Sign in should fail since the user should no longer exist in the user pool.");
+        } catch (Exception e) {
+            assertTrue(e instanceof com.amazonaws.services.cognitoidentityprovider.model.UserNotFoundException);
+        }
+    }
+    
+    @Test(expected = com.amazonaws.mobileconnectors.cognitoidentityprovider.exceptions.CognitoNotAuthorizedException.class)
+    public void testDeleteUserUnauthenticatedUser() throws Exception {
```

Comment:
I wonder if `testDeleteUnauthenticatedUser` might be clearer language?  I imagine you're phrasing it like this so that `deleteUser` is one block

Reply:
I did phrase it as `testDeleteUserUnauthenticatedUser` to make it clear that `deleteUser` is being tested. I agree it sounds a bit clunky. I can rename the test to be `testDeleteUserWithUnauthenticatedUser` which is more consistent with the naming of other tests in this file.